### PR TITLE
[ACR] `az acr login`: Add additional parameter validation to check registry name is valid

### PR DESCRIPTION
**Pull request by mabenedi**
**Related command**
`az acr login`

**Description**

Adds additional parameter validation for `az acr login`

**Testing Guide**

`az acr login -n myregistry` -> good
`az acr login -n myregistry$` -> throw error

**History Notes**

[ACR] `az acr login`: Add additional parameter validation to check registry name is valid

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py [hunk 1]
```diff
@@ -111,6 +113,10 @@ def validate_registry_name(cmd, namespace):
         if pos > 0:
             logger.warning("The login server endpoint suffix '%s' is automatically omitted.", acr_suffix)
             namespace.registry_name = registry[:pos]
+    registry = namespace.registry_name
+    pattern = r'^[a-zA-Z0-9]{5,50}$'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acr/_validators.py:117]: Please add a comment to explain the purpose of this regular expression

**Comment by zhoxing-ms** [comment]
Could you have other engineers in your team review and sign off this PR as well? @mabenedi

**Comment by zhoxing-ms** [comment]
Please note that we are launching the release for this sprint this week. Please resolve all comments by xxx, otherwise the release of this PR will have to be postponed to the next sprint **(on 12-05)**

**Review by SakoPak — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py [hunk 2]
```diff
@@ -111,6 +113,11 @@ def validate_registry_name(cmd, namespace):
         if pos > 0:
             logger.warning("The login server endpoint suffix '%s' is automatically omitted.", acr_suffix)
             namespace.registry_name = registry[:pos]
+    registry = namespace.registry_name
+    # Regex pattern to validate that registry name is alphanumeric and between 5 and 50 characters
+    pattern = r'^[a-zA-Z0-9]{5,50}$'
```
Comment (rosanch) [inline src/azure-cli/azure/cli/command_modules/acr/_validators.py:118]: can you move this to the constants file please?

**Review by rosanch — CHANGES_REQUESTED** [review]
Other than that, one comment LGTM

**Review by rosanch — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**